lib/libmpa/mpa_random.c

```diff
@@ -35,12 +35,12 @@
 #include "mpa_assert.h"
 #include <tee_api_types.h>
 
-/* This function ends to the real RNG of the platform
- * Depending where the libmpa is used:
- * - kernel side: the implementation is found in the TEECore
- * - user side: the implementation is found in libutee
- */
-TEE_Result get_rng_array(void *buf, size_t blen);
+static random_generator_cb get_rng_array;
+
+void mpa_set_random_generator(random_generator_cb callback)
+{
+	get_rng_array = callback;
+}
```

Comment:
Yes, we don't have PRNG implemented in user space. libutee also provides get_prng_array()  which issues a system call (refer to lib/libutee/arch/arm32/utee_misc.c). That is why the trick in the mpa_random.c works. If we call mpa_set_random_generator(get_random_array) properly at TA's initialization routine. We can make sure request random number from user space will trigger a system call.